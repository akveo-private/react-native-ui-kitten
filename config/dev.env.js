const path = require('path');

const evaPath = path.resolve(__dirname, '../../eva');

module.exports = {
  ENV: 'dev',
  KITTEN_PATH: path.resolve(__dirname, '../src/framework'),
  MAPPING_PATH: path.resolve(evaPath, 'packages/mapping/eva'),
  THEME_PATH: path.resolve(evaPath, 'packages/theme/eva'),
};
