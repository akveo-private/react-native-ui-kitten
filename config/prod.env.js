const path = require('path');

const playgroundPath = path.resolve(__dirname, '../src/playground');

module.exports = {
  ENV: 'prod',
  KITTEN_PATH: path.resolve(__dirname, '../src/framework'),
  MAPPING_PATH: path.resolve(playgroundPath, 'node_modules/@eva/eva'),
  THEME_PATH: path.resolve(playgroundPath, 'node_modules/@eva/theme-eva'),
};
