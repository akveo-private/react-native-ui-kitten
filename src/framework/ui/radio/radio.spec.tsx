import React from 'react';
import { TouchableOpacity } from 'react-native';
import {
  render,
  fireEvent,
  waitForElement,
  shallow,
  RenderAPI,
} from 'react-native-testing-library';
import {
  styled,
  StyleProvider,
  StyleProviderProps,
} from '@kitten/theme';
import {
  Radio as RadioComponent,
  Props as RadioProps,
} from './radio.component';
import * as config from './radio.spec.config';
import { ReactTestInstance } from 'react-test-renderer';

const Radio = styled<RadioComponent, RadioProps>(RadioComponent);

const Mock = (props?: RadioProps): React.ReactElement<StyleProviderProps> => (
  <StyleProvider mapping={config.mapping} theme={config.theme} styles={{}}>
    <Radio {...props} />
  </StyleProvider>
);

const renderComponent = (props?: RadioProps): RenderAPI => {
  return render(
    <Mock {...props}/>,
  );
};

describe('@radio: matches snapshot', () => {

  it('default', () => {
    const component: RenderAPI = renderComponent();
    const { output } = shallow(component.getByType(RadioComponent));

    expect(output).toMatchSnapshot();
  });

  it('checked', () => {
    const component: RenderAPI = renderComponent({ checked: true });
    const { output } = shallow(component.getByType(RadioComponent));

    expect(output).toMatchSnapshot();
  });

  it('disabled', () => {
    const component: RenderAPI = renderComponent({ disabled: true });
    const { output } = shallow(component.getByType(RadioComponent));

    expect(output).toMatchSnapshot();
  });

  it('checked disabled', () => {
    const component: RenderAPI = renderComponent({
      checked: true,
      disabled: true,
    });
    const { output } = shallow(component.getByType(RadioComponent));

    expect(output).toMatchSnapshot();
  });

  it('active', async () => {
    const component: RenderAPI = renderComponent();

    fireEvent(component.getByType(TouchableOpacity), 'pressIn');

    const active: ReactTestInstance = await waitForElement(() => {
      return component.getByType(RadioComponent);
    });
    const { output: activeOutput } = shallow(active);

    fireEvent(component.getByType(TouchableOpacity), 'pressOut');

    const inactive: ReactTestInstance = await waitForElement(() => {
      return component.getByType(RadioComponent);
    });
    const { output: inactiveOutput } = shallow(inactive);

    expect(activeOutput).toMatchSnapshot();
    expect(inactiveOutput).toMatchSnapshot('default');
  });

  it('active checked', async () => {
    const component: RenderAPI = renderComponent({ checked: true });

    fireEvent(component.getByType(TouchableOpacity), 'pressIn');

    const active: ReactTestInstance = await waitForElement(() => {
      return component.getByType(RadioComponent);
    });
    const { output: activeOutput } = shallow(active);

    fireEvent(component.getByType(TouchableOpacity), 'pressOut');

    const inactive: ReactTestInstance = await waitForElement(() => {
      return component.getByType(RadioComponent);
    });
    const { output: inactiveOutput } = shallow(inactive);

    expect(activeOutput).toMatchSnapshot();
    expect(inactiveOutput).toMatchSnapshot('checked');
  });

});

describe('@radio: component checks', () => {

  it('* emits onChange with correct args', () => {
    const onChange = jest.fn();

    const component: RenderAPI = renderComponent({
      checked: false,
      onChange: onChange,
    });

    fireEvent.press(component.getByType(TouchableOpacity));

    expect(onChange).toBeCalledWith(true);
  });

});
