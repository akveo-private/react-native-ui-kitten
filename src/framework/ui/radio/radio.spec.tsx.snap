// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@radio: matches snapshot active 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  dispatch={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#DDE1EB",
          "borderRadius": 30,
          "height": 60,
          "opacity": undefined,
          "position": "absolute",
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#8992A3",
          "borderRadius": 18,
          "borderWidth": 2,
          "height": 36,
          "justifyContent": "center",
          "width": 36,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot active checked 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  checked={true}
  dispatch={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#DDE1EB",
          "borderRadius": 30,
          "height": 60,
          "opacity": undefined,
          "position": "absolute",
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#2541CC",
          "borderRadius": 18,
          "borderWidth": 2,
          "height": 36,
          "justifyContent": "center",
          "width": 36,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#3366FF",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot active checked: checked 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  checked={true}
  dispatch={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 30,
          "height": 60,
          "opacity": undefined,
          "position": "absolute",
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#3366FF",
          "borderRadius": 18,
          "borderWidth": 2,
          "height": 36,
          "justifyContent": "center",
          "width": 36,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#3366FF",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot active: default 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  dispatch={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 30,
          "height": 60,
          "opacity": undefined,
          "position": "absolute",
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#A6AEBD",
          "borderRadius": 18,
          "borderWidth": 2,
          "height": 36,
          "justifyContent": "center",
          "width": 36,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot checked 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  checked={true}
  dispatch={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 30,
          "height": 60,
          "opacity": undefined,
          "position": "absolute",
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#3366FF",
          "borderRadius": 18,
          "borderWidth": 2,
          "height": 36,
          "justifyContent": "center",
          "width": 36,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#3366FF",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot checked disabled 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  checked={true}
  disabled={true}
  dispatch={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 30,
          "height": 60,
          "opacity": undefined,
          "position": "absolute",
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#DDE1EB",
          "borderRadius": 18,
          "borderWidth": 2,
          "height": 36,
          "justifyContent": "center",
          "width": 36,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#A6AEBD",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot default 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  dispatch={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 30,
          "height": 60,
          "opacity": undefined,
          "position": "absolute",
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#A6AEBD",
          "borderRadius": 18,
          "borderWidth": 2,
          "height": 36,
          "justifyContent": "center",
          "width": 36,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot disabled 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  disabled={true}
  dispatch={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 30,
          "height": 60,
          "opacity": undefined,
          "position": "absolute",
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#DDE1EB",
          "borderRadius": 18,
          "borderWidth": 2,
          "height": 36,
          "justifyContent": "center",
          "width": 36,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
</TouchableOpacity>
`;
